src/datasets/features/features.py

```diff
@@ -1178,6 +1178,29 @@ def list_of_np_array_to_pyarrow_listarray(l_arr: List[np.ndarray], type: pa.Data
         return pa.array([], type=type)
 
 
+def contains_any_np_array(data):
+    if isinstance(data, np.ndarray):
+        return True
+    elif isinstance(data, list):
+        return any(contains_any_np_array(item) for item in data)
+    else:
+        return False
+
+
+def any_np_array_to_pyarrow_listarray(data, type=None):
+    if isinstance(data, np.ndarray):
+        return numpy_to_pyarrow_listarray(data, type=type)
+    elif isinstance(data, list):
+        return list_of_pa_arrays_to_pyarrow_listarray([any_np_array_to_pyarrow_listarray(i, type=type) for i in data])
+
+
+def to_pyarrow_listarray(data, pa_type):
```

Comment:
Can you add type hints and docstrings  please ?

Reply:
Sure! I just wanted the approach to be validated before spending time on docs... ':)